river/LayoutDemand.zig

```diff
@@ -0,0 +1,150 @@
+// This file is part of river, a dynamic tiling wayland compositor.
+//
+// Copyright 2020 - 2021 The River Developers
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+const Self = @This();
+
+const std = @import("std");
+const wlr = @import("wlroots");
+const wayland = @import("wayland");
+const wl = wayland.server.wl;
+const zriver = wayland.server.zriver;
+
+const log = @import("log.zig");
+const util = @import("util.zig");
+
+const Layout = @import("Layout.zig");
+const Box = @import("Box.zig");
+const Server = @import("Server.zig");
+const Output = @import("Output.zig");
+const View = @import("View.zig");
+const ViewStack = @import("view_stack.zig").ViewStack;
+
+const Error = error{ViewDimensionMismatch};
+
+/// Arbitrary timeout
+const timeout = 1000;
+
+serial: u32 = 0,
+views: u32 = 0,
+output: *Output = undefined,
+
+/// Timeout timer
+timer: ?*wl.EventSource = null,
+
+/// Proposed view dimensions
+view_dimensions: std.ArrayList(Box) = undefined,
+
+pub fn init(self: *Self, output: *Output, views: u32) void {
```

Comment:
We don't need to store pointers to the layout demand anywhere, so we can use a more idiomatic signature for init:
```suggestion
pub fn init(output: *Output, views: u32) Self {
```